api/envoy/admin/v2alpha/metrics.proto

```diff
@@ -11,9 +11,12 @@ message SimpleMetric {
     GAUGE = 1;
   }
 
-  // Type of metric represented.
-  Type type = 1;
+  // Name of the metric.
```

Comment:
kept this in this order so that it comes in that order

Reply:
I'm a bit confused here; are you saying the textual ordering in the proto spec dictates the wire format or JSON order? I would have thought the field numbers would dictate this.

Reply:
Yes. I also thought field numbers would dictate that, looks like not. I changed the order in proto like this

    // Type of the metric represented.
    Type type = 1;
    
  // Name of the metric.
  string name = 3;

  // Current metric value.
  uint64 value = 2;

and the corresponding output is
      {
       "type": "COUNTER",
       "name": "rq_error",
       "value": "0"
      }
Kind of weird but that is how it is

Reply:
Sorry. Some thing wrong. Kept the same order as earlier